packages/components/src/utils/unit-values.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { isNil } from 'lodash';
+
+const UNITED_VALUE_REGEX = /^([\d.\-+]*)\s*(fr|cm|mm|Q|in|pc|pt|px|em|ex|ch|rem|lh|vw|vh|vmin|vmax|%|cap|ic|rlh|vi|vb|deg|grad|rad|turn|s|ms|Hz|kHz|dpi|dpcm|dppx)?$/;
+
+/**
+ * Parses a number and unit from a value.
+ *
+ * @param  toParse Value to parse
+ *
+ * @return  The extracted number and unit.
+ */
+export function parseCSSUnitValue(
+	toParse: string | number
+): [ number | undefined, string | undefined ] {
+	if ( isNil( toParse ) ) {
+		return [ undefined, undefined ];
+	}
```

Comment:
When will this be true? If `toParse` can be `null` or `undefined`, shouldn't we update the parameter types to reflect that?
```ts
toParse?: string | number | null
```
In that case, can we just do this instead of using lodash?
```js
if ( toParse == null ) {
```

Reply:
> shouldn't we update the parameter types to reflect that?

Yes.

> can we just do this instead of using lodash?

I'd love to do that but it requires disabling an eslint rule (`eqeqeq`). If I do that then one person complains, if I don't then another does. I'm really not sure where to go from here about it.

Reply:
>  I'd love to do that but it requires disabling an eslint rule (eqeqeq). If I do that then one person complains

To give more context to @diegohaz , I left a [comment](https://github.com/WordPress/gutenberg/pull/33514#discussion_r671398181) about avoiding disabling  eslint rules inline, since I believe that doing so introduces technical debt.

If we agree that using `== null` is a good pattern, then my proposal is to disable the `eslint` rule at a package level (if not at the repository level). Alternatively, if we don't want to use `lodash`, we could use the more verbose `typeof toParse !== undefined && toParse !== null`.

I just think that having a lint rule and not respecting is not a good pattern; I didn't mean to complain and delay work on this PR.

Reply:
@ciampo can you bring that up in the #core-js meeting in WordPress make slack?

Reply:
I removed the null handling from this. Generally G2 handles null values _too often_ and in non-transparent ways. I think we can instead rely on regular NPE errors surfacing when people use the utility incorrectly.

Reply:
> I'd love to do that but it requires disabling an eslint rule (`eqeqeq`). If I do that then one person complains, if I don't then another does. I'm really not sure where to go from here about it.

Got it! Thanks for the context and sorry about that. The `eqeqeq` rule has an especial option just for the `null` case: https://eslint.org/docs/rules/eqeqeq#allow-null. That's so common that I just assumed that was the default or that we were already allowing it.